### PR TITLE
Fail release if an empty yaml is generated

### LINKED ISSUE #498: Releases should fail if the try to publish 0-length files
The last week of `knative-eventing` releases had two yaml files (`gcp-pubsub.yaml` and `natss.yaml`) which were zero-length (due to yaml formatting errors detected by `ko`) -- See https://github.com/knative/eventing/pull/824 for the fix.

While we should fix the `build_release` function to check `ko`'s exit status, it would probably be a good idea for the release script to also fail rather than publish if any of the requested YAML files are length 0 (belt-and-suspenders). I happened to notice this while cutting the 0.4.0 release because `hub` refuses to publish 0-length files to a release (possibly by accident).

```
gsutil ls -l gs://knative-nightly/eventing/previous/v2019*/gcp-
pubsub.yaml
      4216  2019-02-06T09:41:02Z  gs://knative-nightly/eventing/previous/v20190206-b24b2a2b/gcp-pubsub.yaml
      4216  2019-02-07T09:41:43Z  gs://knative-nightly/eventing/previous/v20190207-4d7f5279/gcp-pubsub.yaml
      4216  2019-02-08T09:43:11Z  gs://knative-nightly/eventing/previous/v20190208-01823f1c/gcp-pubsub.yaml
      4216  2019-02-09T09:43:40Z  gs://knative-nightly/eventing/previous/v20190209-70c9b737/gcp-pubsub.yaml
      4216  2019-02-10T09:40:38Z  gs://knative-nightly/eventing/previous/v20190210-70c9b737/gcp-pubsub.yaml
      4216  2019-02-11T09:39:31Z  gs://knative-nightly/eventing/previous/v20190211-70c9b737/gcp-pubsub.yaml
      4386  2019-02-13T09:41:18Z  gs://knative-nightly/eventing/previous/v20190213-f88998b1/gcp-pubsub.yaml
         0  2019-02-14T09:41:02Z  gs://knative-nightly/eventing/previous/v20190214-0321cd08/gcp-pubsub.yaml
         0  2019-02-15T09:41:03Z  gs://knative-nightly/eventing/previous/v20190215-466dd434/gcp-pubsub.yaml
         0  2019-02-16T09:40:48Z  gs://knative-nightly/eventing/previous/v20190216-c3b25ccd/gcp-pubsub.yaml
         0  2019-02-17T09:38:05Z  gs://knative-nightly/eventing/previous/v20190217-c3b25ccd/gcp-pubsub.yaml
         0  2019-02-18T09:39:00Z  gs://knative-nightly/eventing/previous/v20190218-c3b25ccd/gcp-pubsub.yaml
         0  2019-02-19T09:37:41Z  gs://knative-nightly/eventing/previous/v20190219-c3b25ccd/gcp-pubsub.yaml
         0  2019-02-20T09:39:03Z  gs://knative-nightly/eventing/previous/v20190220-3cdb3add/gcp-pubsub.yaml
```

### PR DESCRIPTION
**Pull request by adrcunha**
Fixes #498

### CODE REVIEW

File: scripts/release.sh [hunk 1]
```diff
@@ -383,6 +383,10 @@ function main() {
     abort "error building the release"
   fi
   [[ -z "${YAMLS_TO_PUBLISH}" ]] && abort "no manifests were generated"
+  # Ensure no empty YAML file will be published.
```
Comment (chaodaiG) [inline scripts/release.sh]: Should this be here or few lines below under `if (( PUBLISH_RELEASE)); then` ?

**Comment by chaodaiG** [comment]
/lgtm
/hold

Left a comment, feel free unhold if you don't think it's needed

File: scripts/release.sh (hunk 1 above)
Comment (adrcunha) [reply scripts/release.sh]: Here. An empty yaml is a build error.

**Pull request merged**